### ISSUE TITLE
Adding and editing data handlers

### ISSUE THREAD

**Issue by localjo** [nasa-gibs/worldview#720]
I need to learn how adding and editing data handlers work in order to proceed with work for #132, #169, #260, #636 and #637 

It looks like handlers are accessed with [a function called `getDataHandlerByName()`](https://github.com/nasa-gibs/worldview/blob/master/web/js/data/handler.js#L37). It looks like a handler itself is [a function with the signature `dataHandlerName: (config: Object, model: Object, spec: Object) => dataHandler: Object`](https://github.com/nasa-gibs/worldview/blob/master/web/js/data/handler.js#L186-L197).

I need more information about what makes up the input and output objects, and what the data handler functions are responsible for doing. It looks like there is some [documentation that explains how Data Download works](https://github.com/nasa-gibs/worldview/blob/master/doc/data_download.md), so I'll read that and then use this ticket to ask any follow up questions.

**Comment by localjo** [comment]
Data handlers are configured in the options repo, in `common/config/wv.json/products/{instrument_name}/{product_shortname}.json`. Product shortnames are listed in `common/config/wv.json/layers/{instrument_name}/{layer_id}.json`.

**Comment by localjo** [comment]
Data handlers control the display of granules and swath lines, or the "select data" modal that lists granules. Does the data handler control the content of the "Download Links" modal?

**Comment by localjo** [comment]
To find a product's data in the Worldview UI, add a layer by searching for the title/subtitle that's defined in `common/config/wv.json/layers/{instrument_name}/{layer_id}.json`.

**Comment by localjo** [comment]
Here's my understanding so far;

The data handler objects returned by the named data handler functions looks like they extend the object returned by [`dataHandlerBase(config, model)`](https://github.com/nasa-gibs/worldview/blob/master/web/js/data/handler.js#L58) by adding `_submit` and `_processResults` properties. The `_submit` property is a function with the signature `_submit: (queryOptions: Object) => cmrRequest: Promise`. The `cmrRequest` promise resolves with `data.feed.entry` from the CMR response. The `_processResults` property is a function with the signature `_processResults: (data: Object) => processedResults: Object`. The `processedResults` object has the shape;
```
{
  meta: results.meta, // Metadata object
  granules: newGranules, // Array of granules
  filtered: filteredGranules // Array of granules
}
```

So, to summarize it looks like for my purposes a data handler is an object with a `_submit` function that makes a CMR request based on an options object, and a `_processResults` function which modifies the results to be what we want to use in Worldview.

@rboller can you confirm my understanding? It may be helpful for us to chat about how this works in person if you have time. 🙂

**Comment by localjo** [comment]
CMR endpoints and responses for a product can be found by searching for the product shortname in https://search.earthdata.nasa.gov and clicking on a collection, then on "View collection details" and then "API Endpoints" and "CMR".

CMR API documentation can be found at https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html

**Comment by localjo** [comment]
The `updateGranules` function in `web/js/data/map.js` is responsible for drawing swath lines and granule buttons on the map. https://github.com/nasa-gibs/worldview/blob/master/web/js/data/map.js#L196
